src/utils.rs

```diff
@@ -11,26 +10,23 @@ pub fn distance(a: Coords2D, b: Coords2D) -> f64 {
 
 //Helper function to recreate path once goal is located
 pub fn rewind(start: &Node, closed: &Vec<Node>) -> Vec<Coords2D> {
-    let mut path = Vec::new();
+    let mut path = Vec::with_capacity(closed.len().pow(2) + 1);
+
     path.push(start.position);
+
     let mut parent = start.parent;
     let mut node = start.position;
 
     while parent != node {
         if let Some(step) = closed.iter().find(|x| x.position == parent) {
             let direction = direction(parent, node);
-            let mut next = Coords2D::from((
-                (node.0 as i32 + direction.0) as usize,
-                (node.1 as i32 + direction.1) as usize,
-            ));
+            let mut next = shift(node, direction);
```

Comment:
The operation of moving a node in a direction is done multiple time,s so I extracted that logic into the `shift()` function